### PR TITLE
Added the ability to remember commits and provide their details later

### PR DESCRIPTION
**Pull request by sjernigan**
Travis CI is now passing on my fork.  There's a lot in this PR so feel free to ask questions.  The changes to the README are probably the easiest place to get an overall feel for the changes.  Thanks again for this project.  We'll be making good use of it at ReverbNation.

### CODE REVIEW

File: templates/commit_details.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[GitHub] Commit <%= @commit['id'][0,Lita::Handlers::GithubCommits::SHA_ABBREV_LENGTH]%> committed by <%=@commit['committer'] ? @commit['committer']['name'] : "<unknown>"%> on branch <%=@commit['branch']%> at <%=Time.parse(@commit['timestamp']).getlocal.to_s%> with message
+  '<%=@commit['message']%>'
+<%if !@commit['added'].empty?%>
+Files Added
+  <%=@commit['added'].join("\n  ")%>
+<%end%>
+<%if !@commit['removed'].empty?%>
+Files Removed
+  <%=@commit['removed'].join("\n  ")%>
+<%end%>
```
Comment (webdestroya) [inline templates/commit_details.erb]: Doesn't this add in a bunch of extra newlines when you put the `if` and `end` on new lines? I know it probably doesn't matter for Slack/Hipchat, but this probably would look bad on IRC.

**Comment by webdestroya** [comment]
Would it be too difficult to add a "use condensed response" that says a one liner or something for IRC? 

`Commit xxxxxx by USER: Fixed the blah blah blah .... stuff`

But have it strip newlines and truncate after ~200 characters. Would that be fairly easy for you to add? I think that is my only complaint since I use the bot in Hipchat/Slack/IRC

**Comment by sjernigan** [comment]
I would make an argument for the longer format because you "asked" for the
commit info.  Whereas, I like the shorter notification when it's
automatically pushed.  However, it's your repo so I can make a hipchat
specific template and a shorter IRC one.  In fact, I tried to already make
a HipChat template to get some HTML features but Hipchat can't accept HTML
yet.

I'm not going to submit a Slack template because I have no way to test it.
It shouldn't be hard to copy the Hipchat one I suppose.

Sound good?

....................................................................................

*Steve Jernigan*
*CIO, Co-Founder*
sjernigan@reverbnation.com
Phone:  919.321.2268 x206  | Mobile: 919.371.8260
Mail: P.O. Box 2086, Durham NC 27702
Deliveries: 115 N. Duke Street, Ste. 2A, Durham NC 27701




Join our team. <https://www.youtube.com/watch?v=MHPz0zUNPPc>

**Comment by webdestroya** [comment]
Steve, that sounds great. As soon as you make that I will merge it

**Comment by sjernigan** [comment]
OK, changes all checked in.  I never figured out how to write the rspec
test to exercise the hipchat handler but it works as expected when
connected to hipchat.

....................................................................................

*Steve Jernigan*
*CIO, Co-Founder*
sjernigan@reverbnation.com
Phone:  919.321.2268 x206  | Mobile: 919.371.8260
Mail: P.O. Box 2086, Durham NC 27702
Deliveries: 115 N. Duke Street, Ste. 2A, Durham NC 27701




Join our team. <https://www.youtube.com/watch?v=MHPz0zUNPPc>

**Pull request merged**